toolchain/check/check.cpp

```diff
@@ -376,12 +377,17 @@ auto CheckParseTrees(
     }
   }
 
+  std::string target = !options.target.empty()
+                           ? options.target.str()
+                           : llvm::sys::getDefaultTargetTriple();
```

Comment:
I think this target defaulting logic should not be in check, and should instead live in the caller. (The language server in this case, which in turn should probably be getting the triple from a compilation database or similar eventually.)

It's probably reasonable to instead CHECK-fail here if the target isn't set.

Reply:
Okay, changed back to just being a regular param. It feels like we can just require it be set that way, rather than using this options struct.